photon-ml/src/integTest/scala/com/linkedin/photon/ml/GameModelGenerator.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linkedin.photon.ml
+
+import org.apache.log4j.Logger
+import org.apache.spark.SparkContext
+
+import com.linkedin.photon.ml.TaskType.TaskType
+import com.linkedin.photon.ml.model.{Coefficients, FixedEffectModel, GAMEModel, RandomEffectModel}
+import com.linkedin.photon.ml.supervised.classification.LogisticRegressionModel
+import com.linkedin.photon.ml.supervised.regression.LinearRegressionModel
+
+/**
+ * A generator of Game models.
+ *
+ * The models generated are random, within certain parameters that the user can specify (e.g., number of random effects,
+ * number of features in each random effect, ...).
+ */
+class GameModelGenerator {
+
+  private val log : Logger = Logger.getLogger(classOf[GameModelGenerator])
+  private val rng = scala.util.Random
+
+  private def randInt(M: Int) = 1 + math.abs(rng.nextInt) % M
+
+  private def randSparseVector(M: Int) = {
```

Comment:
Yeah, it's pseudo-uniform, and there are shortcuts. It's a trade-off between ease and being completely correct. I opted for ease on the assumption that this would be used in tests only, where aspects such as the size of the model would matter more than the distribution of the non-zeros over the feature space. I could be wrong about that assumption. The best option would be to let the user specify the distribution of the non-zeros, but I thought that was overkill here.